src/components/Select/Select.tsx

```diff
@@ -0,0 +1,194 @@
+  size,
+  autoUpdate,
+  FloatingPortal,
+  FloatingOverlay,
+} from "@floating-ui/react";
+import styles from "./Select.module.css";
+
+interface Props {
+  options: string[];
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+  defaultSelected?: number;
+  renderHandler?: (value: string) => React.ReactNode | undefined;
+  renderValue?: (value: string) => React.ReactNode | undefined;
+  onChange?: (index: number) => void;
+}
+
+export function Select({
+  options,
+  isOpen,
+  setIsOpen,
+  defaultSelected,
+  renderValue,
+  renderHandler,
+  onChange,
+}: Props) {
+  const [activeIndex, setActiveIndex] = React.useState<number | null>(null);
+  const [selectedIndex, setSelectedIndex] = React.useState<number | null>(
+    typeof defaultSelected === "number" ? defaultSelected : null
+  );
+
+  const { x, y, strategy, refs, context } = useFloating({
+    placement: "bottom-start",
+    open: isOpen,
+    onOpenChange: setIsOpen,
+    whileElementsMounted: autoUpdate,
+    middleware: [
+      // offset(5),
+      flip({ padding: 10 }),
+      size({
+        apply({ rects, elements, availableHeight }) {
+          Object.assign(elements.floating.style, {
+            maxHeight: `${availableHeight}px`,
+            width: `${rects.reference.width}px`,
+          });
+        },
+        padding: 10,
+      }),
+    ],
+  });
+
+  const listRef = React.useRef<Array<HTMLElement | null>>([]);
+  const listContentRef = React.useRef(options);
+  const isTypingRef = React.useRef(false);
+
+  const click = useClick(context, { event: "mousedown" });
+  const dismiss = useDismiss(context);
+  const role = useRole(context, { role: "listbox" });
+  const listNav = useListNavigation(context, {
+    listRef,
+    activeIndex,
+    selectedIndex,
+    onNavigate: setActiveIndex,
+    // This is a large list, allow looping.
+    loop: true,
+  });
+  const typeahead = useTypeahead(context, {
+    listRef: listContentRef,
+    activeIndex,
+    selectedIndex,
+    onMatch: isOpen ? setActiveIndex : setSelectedIndex,
+    onTypingChange(isTyping) {
+      isTypingRef.current = isTyping;
+    },
+  });
+
+  const { getReferenceProps, getFloatingProps, getItemProps } = useInteractions(
+    [click, dismiss, role, listNav, typeahead]
+  );
+
+  const handleSelect = (index: number) => {
+    setSelectedIndex(index);
+    setIsOpen(false);
+    onChange && onChange(index);
+  };
+
+  const selectedItemLabel =
+    selectedIndex !== null ? options[selectedIndex] : undefined;
+
+  const handler =
+    renderHandler && selectedItemLabel
+      ? renderHandler(selectedItemLabel)
+      : null;
+
+  return (
+    <>
+      <div
+        tabIndex={0}
+        ref={refs.setReference}
+        aria-autocomplete="none"
+        className={styles.select}
+        {...getReferenceProps()}
+      >
+        {handler || "Select..."}
+      </div>
+
+      <FloatingPortal>
+        {isOpen && (
+          <FloatingOverlay style={{ zIndex: 2 }} lockScroll={isOpen}>
```

Comment:
Maybe we'll delete `lockScroll` and `overflow: hidden` on dropdown open?

The page jumps :(

Reply:
removed